usage_test.go

```diff
@@ -459,6 +459,170 @@ func TestNonexistentSubcommand(t *testing.T) {
 	assert.Error(t, err)
 }
 
+func TestUsageWithOptionsGroup(t *testing.T) {
+	expectedUsage := "Usage: example [--verbose] [--insecure] [--host HOST] [--port PORT] [--user USER] OUTPUT"
+
+	expectedHelp := `
+Usage: example [--verbose] [--insecure] [--host HOST] [--port PORT] [--user USER] OUTPUT
+
+Positional arguments:
+  OUTPUT
+
+Options:
+  --verbose, -v          verbosity level
+
+Database options:
+
+This block represents related arguments.
+
+  --insecure, -i         disable tls
+  --host HOST            hostname to connect to [default: localhost, env: DB_HOST]
+  --port PORT            port to connect to [default: 3306, env: DB_PORT]
+  --user USER            username to connect as [env: DB_USERNAME]
+  (environment only)     password to connect with [env: DB_PASSWORD]
+
+Global options:
+  --help, -h             display this help and exit
+`
+
+	type database struct {
+		Insecure bool   `arg:"-i,--insecure" help:"disable tls"`
+		Host     string `arg:"--host,env:DB_HOST" help:"hostname to connect to" default:"localhost"`
+		Port     string `arg:"--port,env:DB_PORT" help:"port to connect to" default:"3306"`
+		User     string `arg:"--user,env:DB_USERNAME" help:"username to connect as"`
+		Password string `arg:"--,env:DB_PASSWORD" help:"password to connect with"`
+	}
+
+	var args struct {
+		Verbose  bool      `arg:"-v" help:"verbosity level"`
+		Database *database `arg:"group" help:"This block represents related arguments."`
+		Output   string    `arg:"positional,required"`
+	}
+
+	os.Args[0] = "example"
+	p, err := NewParser(Config{}, &args)
+	require.NoError(t, err)
+
+	_ = p.Parse([]string{})
+
+	var help bytes.Buffer
+	p.WriteHelp(&help)
+	assert.Equal(t, expectedHelp[1:], help.String())
+
+	var usage bytes.Buffer
+	p.WriteUsage(&usage)
+	assert.Equal(t, expectedUsage, strings.TrimSpace(usage.String()))
+}
+
+func TestUsageWithSubcommandAndOptionsGroup(t *testing.T) {
+	testCases := []struct {
```

Comment:
Can you keep the current convention of a separate top-level test function for each test case? I know it's a little more verbose and that it's common to have test cases in Go set up in the way that you've set them up, but I personally really appreciate the directness of separate top-level test functions for each test case.

Reply:
A bit WET :)
But I've unrolled the table tests into separate test functions 👍🏻